src/lib/parser.c

```diff
@@ -1664,6 +1674,191 @@ RdbStatus elementSetLP(RdbParser *p) {
     return nextParsingElement(p, PE_END_KEY);
 }
 
+RdbStatus elementZset(RdbParser *p) {
+    ElementCtx *ctx = &p->elmCtx;
+    enum ZSET_STATES {
+        ST_ZSET_HEADER=0, /*  Retrieve number of nodes */
+        ST_ZSET_NEXT_ITEM /* Process next node and callback to app (Iterative) */
+    };
+    switch (ctx->state) {
+        case ST_ZSET_HEADER:
+            IF_NOT_OK_RETURN(rdbLoadLen(p, NULL, (uint64_t *) &(ctx->key.numItemsHint), NULL, NULL));
+
+            /*** ENTER SAFE STATE ***/
+
+            ctx->zset.left = ctx->key.numItemsHint;
+
+            updateElementState(p, ST_ZSET_NEXT_ITEM); /* fall-thru */
+
+        case ST_ZSET_NEXT_ITEM: {
+            double score;
+            BulkInfo *binfoItem;
+
+
+            IF_NOT_OK_RETURN(rdbLoadString(p, RQ_ALLOC_APP_BULK, NULL, &binfoItem));
+
+            if (p->currOpcode == RDB_TYPE_ZSET_2) {
+                IF_NOT_OK_RETURN(rdbLoadDoubleValue(p, &score));
+            } else {
+                BulkInfo *binfoItem2;
+                IF_NOT_OK_RETURN(rdbLoadString(p, RQ_ALLOC_APP_BULK, NULL, &binfoItem2));
+
+                if (!string2d((const char*) binfoItem2->ref, binfoItem2->len, &score)) {
+                    RDB_reportError(p, RDB_ERR_PLAIN_ZSET_INVALID_STATE,
+                                    "elementZset(): failed to parse string to double: %.*s",
+                                    binfoItem2->len, binfoItem2->ref);
+
+                    return RDB_STATUS_ERROR;
+                }
+            }
+
+            /*** ENTER SAFE STATE ***/
+
+            registerAppBulkForNextCb(p, binfoItem);
+            if (p->elmCtx.key.handleByLevel == RDB_LEVEL_STRUCT)
+                CALL_HANDLERS_CB(p, NOP, RDB_LEVEL_STRUCT, rdbStruct.handleZsetPlain, binfoItem->ref, score);
+            else
+                CALL_HANDLERS_CB(p, NOP, RDB_LEVEL_DATA, rdbData.handleZsetMember, binfoItem->ref, score);
+
+            return (--ctx->zset.left) ? updateElementState(p, ST_ZSET_NEXT_ITEM) : nextParsingElement(p, PE_END_KEY);
+        }
+        default:
+            RDB_reportError(p, RDB_ERR_PLAIN_ZSET_INVALID_STATE,
+                            "elementZset(): invalid parsing element state: %d", ctx->state);
+            return RDB_STATUS_ERROR;
+    }
+}
+
+/* return either RDB_STATUS_OK or RDB_STATUS_ERROR */
+static RdbStatus zsetZiplistItem(RdbParser *p, BulkInfo *ziplistBulk) {
+
+    int ret = ziplistValidateIntegrity(ziplistBulk->ref, ziplistBulk->len, p->deepIntegCheck, NULL, NULL);
```

Comment:
please move the location of zsetZiplistItem() upward to be between `/*** various functions ***/` and `/*** Parsing Common Elements ***/`

Reply:
done